### PR TITLE
Added SDK Name and SDK version to Java SDK

### PR DESCRIPTION
**Pull request by michaelmaxyz**
Additionally, added applicationName and machineName as possible fields in the header as well. Added getter in API and the option to use applicationName in LockstepAPI.

### CODE REVIEW

File: src/main/java/io/lockstep/api/RestRequest.java [hunk 1]
```diff
@@ -145,16 +146,32 @@ public void AddBody(Object body)
...
+            request.addHeader("SdkName", "Java");
+            request.addHeader("SdkVersion", "2022.3.32.0");
             
+            String applicationName = this.client.getAppName();
+
+            if (applicationName != null) {
+                request.addHeader("ApplicationName", applicationName);
+            }
+
+            String machineName = InetAddress.getLocalHost().getHostName();
```
Comment (tspence) [inline src/main/java/io/lockstep/api/RestRequest.java:159]: I've heard some reports that this approach may not be ideally portable, but I am not certain there's a better solution.  Here's what I've found:

https://stackoverflow.com/questions/7348711/recommended-way-to-get-hostname-in-java

The best advice I can seem to find is this one:

```
String hostname = new BufferedReader(
    new InputStreamReader(Runtime.getRuntime().exec("hostname").getInputStream()))
   .readLine();
```

Do you think we should consider doing that once when the client is constructed, and then caching it?  Or would you prefer to keep the InetAddress approach in place?  I'm not completely certain which is better.

File: src/main/java/io/lockstep/api/RestRequest.java [hunk 2]
```diff
@@ -145,16 +146,32 @@ public void AddBody(Object body)
...
+                request.addHeader("Authorization", "Bearer " + bearerToken);
+            }
+
             // Add authentication
             String apiKey = this.client.getApiKey();
             if (apiKey != null) {
                 request.addHeader("Api-Key", apiKey);
             }
-            String bearerToken = this.client.getBearerToken();
-            if (apiKey != null) {
```
Comment (tspence) [inline src/main/java/io/lockstep/api/RestRequest.java:155]: oh good catch

File: src/main/java/io/lockstep/api/RestRequest.java (hunk 1 above)
Comment (michaelmaxyz) [reply src/main/java/io/lockstep/api/RestRequest.java:159]: I think we should use the current approach.

**Comment by tspence** [comment]
LGTM!

**Pull request merged**